query-planner-js/src/buildPlan.ts

```diff
@@ -2751,6 +2745,42 @@ function computePlanInternal({
     ({ main, deferred } = dependencyGraph.process(processor, operation.rootKind));
     primarySelection = dependencyGraph.deferTracking.primarySelection;
 
+    // If this is a subscription, we want to make sure that we return a SubscriptionPlanNode rather than a PlanNode
```

Comment:
I wouldn't have put this code here. Instead, I would have continued to return a `PlanNode` here, and instead deal with transforming it for subscriptions at the end of `buildQueryPlan`. Conceptually, the subscription handling here is a top-level concern: we generate plans as for queries, but then extract the primary fetch separately. So having that change done in `computePlanInternal` feels like the wrong place, and in practice this shows up in the added checks in `computePlanForDeferConditionals` which shouldn't be necessary.